helperAPI.py

```diff
@@ -338,47 +338,71 @@ def get_result(self):
         return self.queue.get()
 
 
+def is_up_to_date(remote, branch):
+    # Assume succeeded in updater()
+    import git
+
+    # Check if local branch is up to date with ls-remote
+    up_to_date = False
+    remote_hash = ""
+    try:
+        g = git.cmd.Git()
+        ls_remote = g.ls_remote(remote, branch)
+        remote_hash = ls_remote.split("\t")[0]
+        local_commit = git.Repo(".").head.commit.hexsha
+        if str(local_commit) == str(remote_hash):
+            up_to_date = True
+            print(f"You are up to date with {remote}/{branch}")
+    except Exception as e:
+        print(f"Error running ls-remote: {e}")
+    if not up_to_date:
+        if remote_hash == "":
+            remote_hash = "NOT FOUND"
+        print(
+            f"WARNING: YOU ARE OUT OF DATE. Please run 'git pull {remote} {branch}' to update. Local hash: {local_commit}, Remote hash: {remote_hash}"
```

Comment:
If we run into a situation where `ls_remote = g.ls_remote(remote, branch)` on line 350 raises an error, we will have a situation where local_commit will be referenced before initiation. This situation should be handled properly.

Reply:
I moved this above the `ls-remote` try-catch:
```python 
local_commit = git.Repo(".").head.commit.hexsha
```
It should be fine since the `updater()` function ensures that the git repository is set up correctly by the time this is run.

I also moved/removed some string casts since `hexsha` is already a string and `remote_hash` is always treated as a string so I cast it at initialization instead of in the comparison.